mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -520,6 +573,24 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	return nil
 }
 
+// getExternalAccountConfig returns an externalaccount.Config based on the Config object.
+// The external account config is intended to be used to create a token source.
+func (c *Config) getExternalAccountConfig(clientScopes []string) externalaccount.Config {
+	eaConfig := externalaccount.Config{
+		Audience:         c.ExternalCredentials.Audience,
+		SubjectTokenType: "urn:ietf:params:oauth:token-type:jwt",
+		ServiceAccountImpersonationURL: fmt.Sprintf("https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/%s:generateAccessToken", c.ExternalCredentials.ServiceAccountEmail),
+		Scopes:               clientScopes,
+		SubjectTokenSupplier: c.ExternalCredentials,
+	}
+	// If UniverseDomain is set, the externalaccount package will use it to set the TokenURL (https://sts.UNIVERSE_DOMAIN/v1/token). Otherwise TokenURL defaults to https://sts.googleapis.com/v1/token
```

Comment:
Appreciate the universe domain addition here - I am curious how you recognized the need for it. I'm sure you were not able to test a case of external credentials + universe domain?

Reply:
This came from looking into the GetCredentials function, which returns a Credentials struct that reflects whether the provider is set with credentials/access_token/ADCs, whether impersonation should be used, and to allow the use of a Universe Domain or not.

the function then returns the credential to create a token source.